src/cryptography/x509.py

```diff
@@ -29,6 +31,77 @@ def __init__(self, msg, parsed_version):
         self.parsed_version = parsed_version
 
 
+class UnknownAttribute(Exception):
+    pass
+
+
+class Attribute(object):
+    def __init__(self, oid, value):
+        if not isinstance(oid, ObjectIdentifier):
+            raise TypeError("oid argument must be an ObjectIdentifier object")
+
+        self._oid = oid
+        self._value = value
+
+    oid = utils.read_only_property("_oid")
+    value = utils.read_only_property("_value")
+
+    def __eq__(self, other):
+        if not isinstance(other, Attribute):
+            return NotImplemented
+
+        return (
+            self.oid == other.oid and
+            self.value == other.value
+        )
+
+    def __ne__(self, other):
+        return not self == other
+
+
+class ObjectIdentifier(object):
+    def __init__(self, oid, name):
+        self._value = oid
+        self._name = name
```

Comment:
Err, it would fallback sanely:

```py
def __repr__(self):
    return "<ObjectIdentifier(oid={0}, name={1}>".format(oid=self._value, name=_OID_NAMES.get(self._value, "Unknown OID"))
```